src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -4079,6 +4080,115 @@ void CalcISO15099WindowIntConvCoeff(EnergyPlusData &state,
+                    });
+                    if (it2 == std::end(uniqueRoofVertices)) {
+                        uniqueRoofVertices.emplace_back(curVertex);
+                    }
+                }
+
+                SurfaceGeometry::EdgeOfSurf thisEdge;
+                thisEdge.start = curVertex;
+                thisEdge.end = nextVertex;
+                thisEdge.count = 1;
+
+                // Uses the custom operator== that uses isAlmostEqual3dPt internally and doesn't care about order of the start/end
+                auto itEdge = std::find(uniqEdgeOfSurfs.begin(), uniqEdgeOfSurfs.end(), thisEdge);
+                if (itEdge == uniqEdgeOfSurfs.end()) {
+                    uniqEdgeOfSurfs.emplace_back(std::move(thisEdge));
+                } else {
+                    ++(itEdge->count);
+                }
+            }
+        }
+    }
+
+    RoofGeo.Height /= RoofGeo.Area;
+    RoofGeo.Tilt /= RoofGeo.Area;
+    RoofGeo.Azimuth /= RoofGeo.Area;
+
+    // Remove the ones that are already used twice
+    uniqEdgeOfSurfs.erase(std::remove_if(uniqEdgeOfSurfs.begin(), uniqEdgeOfSurfs.end(), [](const auto &edge) -> bool { return edge.count == 2; }),
+                          uniqEdgeOfSurfs.end());
+
+    // Intersect with unique vertices as much as needed
+    bool insertedVertext = true;
+    while (insertedVertext) {
+        // Use an index-based loop because I may emplace_back inside the loop, and that invalidates the iterators
+        for (size_t i = 0; i < uniqEdgeOfSurfs.size(); ++i) {
+
+            auto &edge = uniqEdgeOfSurfs[i];
+
+            insertedVertext = false;
+
+            // now go through all the vertices and see if they are colinear with start and end vertices
+            for (const auto &testVertex : uniqueRoofVertices) {
+                if (edge.containsPoints(testVertex)) {
+                    SurfaceGeometry::EdgeOfSurf newEdgeOfSurface;
+                    newEdgeOfSurface.start = testVertex;
+                    newEdgeOfSurface.end = edge.end;
+                    edge.end = testVertex;
+                    uniqEdgeOfSurfs.emplace_back(std::move(newEdgeOfSurface));
+                    break;
+                }
+            }
+            // Break out of the loop on edges, and start again at the while
+            if (insertedVertext) {
+                break;
+            }
+        }
+    }
+
+    // recount
+    for (auto &edge : uniqEdgeOfSurfs) {
+        edge.count = std::count(uniqEdgeOfSurfs.begin(), uniqEdgeOfSurfs.end(), edge);
+    }
+
+    uniqEdgeOfSurfs.erase(std::remove_if(uniqEdgeOfSurfs.begin(), uniqEdgeOfSurfs.end(), [](const auto &edge) -> bool { return edge.count == 2; }),
+                          uniqEdgeOfSurfs.end());
+
+    RoofGeo.Perimeter =
+        std::accumulate(uniqEdgeOfSurfs.cbegin(), uniqEdgeOfSurfs.cend(), 0.0, [](const double &sum, const SurfaceGeometry::EdgeOfSurf &edge) {
+            return sum + edge.length();
+        });
```

Comment:
Now that we only have unique edges that aren't shared, calculate the total of each edge's length: that's your actual perimeter.